packages/memory-inspector/src/browser/memory-widget/memory-options-widget.tsx

```diff
@@ -0,0 +1,726 @@
+                        title={nls.localize('theia/memory-inspector/offsetTooltip', 'Offset to be added to the current memory location, when navigating')}
+                        defaultValue='0'
+                        passRef={this.assignOffsetRef}
+                        onKeyDown={this.doRefresh}
+                        disabled={!this.doUpdateAutomatically}
+                    />
+                    <MWInput
+                        id={LENGTH_FIELD_ID}
+                        label={nls.localize('theia/memory-inspector/length', 'Length')}
+                        title={nls.localize('theia/memory-inspector/lengthTooltip', 'Number of bytes to fetch, in decimal or hexadecimal')}
+                        defaultValue={this.readLength.toString()}
+                        passRef={this.assignReadLengthRef}
+                        onChange={Utils.validateNumericalInputs}
+                        onKeyDown={this.doRefresh}
+                        disabled={!this.doUpdateAutomatically}
+                    />
+                    <button
+                        type='button'
+                        className='theia-button main view-group-go-button'
+                        onClick={this.doRefresh}
+                        disabled={!this.doUpdateAutomatically}
+                        title={nls.localizeByDefault('Go')}
+                    >
+                        {nls.localizeByDefault('Go')}
+                    </button>
+                </div>
+                <div className={`t-mv-memory-fetch-error${this.showMemoryError ? ' show' : ' hide'}`}>
+                    {this.memoryReadError}
+                </div>
+            </>
+        );
+    }
+
+    protected activateHeaderInputField = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        if (!this.isTitleEditable) {
+            const isMouseDown = !('key' in e);
+            const isActivationKey = 'key' in e && (
+                KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.SPACE.keyCode
+                || KeyCode.createKeyCode(e.nativeEvent).key?.keyCode === Key.ENTER.keyCode
+            );
+            if (isMouseDown || isActivationKey) {
+                if (isMouseDown) {
+                    e.currentTarget.blur();
+                }
+                this.isTitleEditable = true;
+                this.update();
+            }
+        }
+    };
+
+    protected saveHeaderInputValue = (e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>): void => {
+        const isMouseDown = !('key' in e);
+        const isSaveKey = 'key' in e && e.key === 'Enter';
+        const isCancelKey = 'key' in e && e.key === 'Escape';
+        e.stopPropagation();
+        if (isMouseDown || isSaveKey || isCancelKey) {
+            if (!isCancelKey && this.headerInputField) {
+                this.title.label = this.headerInputField.value;
```

Comment:
Would it be possible to move this into a separate protected method (like `updateHeader`) so that it is easier to hook into that for some custom behavior?

Reply:
Done.